protostar/self/cache_io.py

```diff
@@ -10,8 +10,13 @@ class CacheIO:
     def __init__(self, project_root_path: Path):
         self._cache_path = project_root_path / Path(self._CACHE_DIR_NAME)
         self._cache_path.mkdir(exist_ok=True)
+        self._gitignore_path = Path(self._cache_path / ".gitignore")
+        self._gitignore_checked = False
 
     def write(self, name: str, value: dict) -> None:
+        if self._gitignore_checked and not self._gitignore_path.exists():
```

Comment:
shouldn't this look like that?

```suggestion
        if not self._gitignore_checked and not self._gitignore_path.exists():
```